cyoung/stratux
# Improve ownship detection for EFBs that care

**Description**
Foreflight was recently updated to improve ownship detection for users with ADS-B out equipped aircraft by comparing gps/baro information against the reported traffic, allowing it to determine the N-Number of the aircraft for track log recording and automatic draft logbook entries.  This update affected Stratux users with aircraft not equipped with ADS-B out, as Foreflight would report 'Ownship Not Detected' even if an ICAO address was set in Stratux configuration.  A recent update to Stratux fixed this issue, however, it negatively affected Stratux users with ADS-B equipped aircraft as it would require them to update the ICAO address in Stratux for proper N-Number detection, or be subjected to draft logbook entries reporting aircraft as 'Stratux'.  This update attempts to deliver the best of both worlds by setting the appropriate traffic type flag based on ICAO Mode-S code defined in Stratux.  A dummy or blank ICAO Mode-S will result in Foreflight's ownship detection logic being used.  A configured ICAO Mode-S setting will result in Stratux passing the ownship information and Foreflight consuming it as is.  Users of other EFB apps such as Aerovie, FlyQ, etc are not affected by this or previous change. 

**What Is Changing**
Update Ownship target identity to report 0x00 which represents 'No Traffic Alert' and 'ADS-B with ICAO' only when a valid ICAO code is configured in Stratux Settings.  Otherwise, send 0x01 for 'No Traffic Alert' and 'ADS-B with Self-Assigned Code'.

**Pre-Testing - Non-Assigned Code**
* Assign ICAO code of F00000 or 000000 in Mode S Code configuration input box
* Open Foreflight EFB
* Go to Devices -> Stratux
* Verify Ownship detection reports 'Stratux' for Aircraft N-Number.

**Pre-Testing - Assigned Code**
* Assign valid ICAO code in Stratux Mode S Code configuration input box
* Open Foreflight EFB
* Go to Devices -> Stratux
* Verify Ownship detection reports 'Detected' and correct N-Number of ICAO entry

**Post-Testing - Unassigned Code**
* Assign ICAO code of F00000 or 000000 in Mode S Code configuration input box
* Open Foreflight EFB
* Go to Devices -> Stratux
* Verify Ownship detection reports 'Not Detected'

**Post-Testing - Assigned Code**
* Assign valid ICAO code in Stratux Mode S Code configuration input box
* Open Foreflight EFB
* Go to Devices -> Stratux
* Verify Ownship detection reports 'Detected' and correct N-Number of ICAO entry
